skywalking-collector/skywalking-collector-worker/src/main/java/com/a/eye/skywalking/collector/worker/application/ApplicationMember.java

```diff
@@ -0,0 +1,108 @@
+import com.a.eye.skywalking.collector.actor.selector.RollingSelector;
+import com.a.eye.skywalking.collector.worker.application.metric.TraceSegmentRecordMember;
+import com.a.eye.skywalking.collector.worker.application.persistence.DAGNodePersistence;
+import com.a.eye.skywalking.collector.worker.application.persistence.NodeInstancePersistence;
+import com.a.eye.skywalking.collector.worker.application.persistence.ResponseCostPersistence;
+import com.a.eye.skywalking.collector.worker.application.persistence.ResponseSummaryPersistence;
+import com.a.eye.skywalking.trace.Span;
+import com.a.eye.skywalking.trace.TraceSegment;
+import com.a.eye.skywalking.trace.tag.Tags;
+
+/**
+ * @author pengys5
+ */
+public class ApplicationMember extends AbstractMember {
+
+    public ApplicationMember(MemberSystem memberSystem, ActorRef actorRef) {
+        super(memberSystem, actorRef);
+    }
+
+    @Override
+    public void preStart() throws Throwable {
+        TraceSegmentRecordMember.Factory factory = new TraceSegmentRecordMember.Factory();
+        factory.createWorker(memberContext(), getSelf());
+    }
+
+    @Override
+    public void receive(Object message) throws Throwable {
+        if (message instanceof TraceSegment) {
+            TraceSegment traceSegment = (TraceSegment) message;
+            AbstractMember discoverMember = memberContext().memberFor(TraceSegmentRecordMember.class.getSimpleName());
+            discoverMember.receive(traceSegment);
+
+            sendToDAGNodePersistence(traceSegment);
+            sendToNodeInstancePersistence(traceSegment);
+            sendToResponseCostPersistence(traceSegment);
+            sendToResponseSummaryPersistence(traceSegment);
+        }
+    }
+
+    public static class Factory extends AbstractMemberProvider {
+        @Override
+        public Class memberClass() {
+            return ApplicationMember.class;
+        }
+    }
+
+    private void sendToDAGNodePersistence(TraceSegment traceSegment) throws Throwable {
+        String code = traceSegment.getApplicationCode();
+
+        String component = null;
+        String layer = null;
+        for (Span span : traceSegment.getSpans()) {
+            if (span.getParentSpanId() == -1) {
+                component = Tags.COMPONENT.get(span);
+                layer = Tags.SPAN_LAYER.get(span);
+            }
+        }
+
+        DAGNodePersistence.Metric node = new DAGNodePersistence.Metric(code, component, layer);
+        tell(new NodeInstancePersistence.Factory(), RollingSelector.INSTANCE, node);
+    }
+
+    private void sendToNodeInstancePersistence(TraceSegment traceSegment) throws Throwable {
+        String code = traceSegment.getPrimaryRef().getApplicationCode();
+        String address = traceSegment.getPrimaryRef().getPeerHost();
+
+        NodeInstancePersistence.Metric property = new NodeInstancePersistence.Metric(code, address);
+        tell(new NodeInstancePersistence.Factory(), RollingSelector.INSTANCE, property);
```

Comment:
The `tell` style is fine. But why need to create a factory instance every time? IMO, factory should stay in singleton.